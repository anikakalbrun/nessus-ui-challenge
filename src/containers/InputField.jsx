import React, {Component} from 'react';
import {connect} from "react-redux";

import {setInputValue} from '../actions';
import {fetchServerConfigurations} from '../actions'

export class InputField extends Component {

    componentDidUpdate(prevProps) {
        if(prevProps.inputValue !== this.props.inputValue) {
            this.props.fetchServerConfigurations('api.tenable.io', this.props.inputValue)
        }
    }

    handleChange = (event) => {
        if(event.target.value.length > 4) return;
        this.props.setInputValue(event.target.value);
    };

    render() {
        const {inputValue} = this.props;
        return (
            <div className="form-group InputField">
                <label className='col-lg-12' style={{padding: 0}}>Enter number of expected records:</label>
                <div className='col-lg-3' style={{padding: 0}}>
                <input
                    name="host"
                    className="form-control"
                    type="number"
                    min={0}
                    max={9999}
                    step={1}
                    onChange={this.handleChange}
                    value={inputValue}
                />
                <small className="form-text text-muted">Table data will be generated</small>
                </div>
            </div>
        );
    }
}

export default connect(state => ({inputValue: state.inputValue}), {setInputValue, fetchServerConfigurations})(InputField);