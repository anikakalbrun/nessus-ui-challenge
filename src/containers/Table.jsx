import React, {Component} from 'react';
import {array} from 'prop-types';
import {connect} from "react-redux";
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
import 'react-bootstrap-table/dist/react-bootstrap-table-all.min.css';

import {fetchServerConfigurations} from '../actions';

const options = {
    sizePerPageList: [{
        text: '5', value: 5
    }, {
        text: '10', value: 10
    }, {
        text: '25', value: 25
    }]
};
const filterType = {type: 'TextFilter', delay: 1000};

export class Table extends Component {

    static propTypes = {
        data: array
    };

    static defaultProps = {
        data: []
    };

    componentWillMount() {
        this.props.fetchServerConfigurations('api.tenable.io', 2)
    }

    render() {
        const {data} = this.props;
        return (
            <BootstrapTable data={data}
                            pagination={true}
                            options={options}
                            hover={true}
                            exportCSV
                            csvFileName={'table-export.csv'}>
                <TableHeaderColumn dataField='name'
                                   filter={filterType}
                                   isKey>Server</TableHeaderColumn>
                <TableHeaderColumn dataField='hostname'
                                   filter={filterType}>Hostname</TableHeaderColumn>
                <TableHeaderColumn dataField='port'
                                   filter={filterType}>Port</TableHeaderColumn>
                <TableHeaderColumn dataField='username'
                                   filter={filterType}>Username</TableHeaderColumn>
            </BootstrapTable>
        );
    }
}

function mapStateToProps(state) {
    return {
        data: state.data.configurations
    };
}

export default connect(mapStateToProps, {fetchServerConfigurations})(Table);