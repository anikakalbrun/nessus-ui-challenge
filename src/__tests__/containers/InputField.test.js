import React from 'react';
import { InputField } from '../../containers/InputField';

let wrapper;
const setInputValue = jest.fn();
const fetchServerConfigurations = jest.fn();
const props = {
    inputValue: '12',
    setInputValue,
    fetchServerConfigurations
};

describe('InputField', () => {

    it('renders correctly', () => {
        wrapper = renderer.create(<InputField {...props}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('input field value change calls setInputValue action', () => {
        wrapper = shallow(<InputField {...props}/>);
        wrapper.find('input').simulate('change', {target: {value: '22'}});
        expect(setInputValue).toHaveBeenCalled()
    });

    it('fetchServerConfigurations action gets called when component receives new props: inputValue', () => {
        wrapper = mount(<InputField {...props}/>);
        wrapper.setProps({ 'inputValue': '10' });
        expect(fetchServerConfigurations).toHaveBeenCalled();
    });
});
