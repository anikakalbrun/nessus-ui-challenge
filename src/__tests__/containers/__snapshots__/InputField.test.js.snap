// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<div
  className="form-group InputField"
>
  <label
    className="col-lg-12"
    style={
      Object {
        "padding": 0,
      }
    }
  >
    Enter number of expected records:
  </label>
  <div
    className="col-lg-3"
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <input
      className="form-control"
      max={9999}
      min={0}
      name="host"
      onChange={[Function]}
      step={1}
      type="number"
      value="12"
    />
    <small
      className="form-text text-muted"
    >
      Table data will be generated
    </small>
  </div>
</div>
`;
